modules/account-lib/src/coin/dot/transaction.ts

```diff
@@ -175,6 +182,94 @@ export class Transaction extends BaseTransaction {
     return result;
   }
 
+  explainTransferTransaction(json: TxData, explanationResult: TransactionExplanation): TransactionExplanation {
+    explanationResult.displayOrder.push('owner', 'forceProxyType');
+    return {
+      ...explanationResult,
+      outputs: [
+        {
+          address: json.to?.toString() || '',
+          amount: json.amount?.toString() || '',
+        },
+      ],
+      owner: json.owner,
+      forceProxyType: json.forceProxyType,
+    };
+  }
+
+  explainStakingActivateTransaction(json: TxData, explanationResult: TransactionExplanation): TransactionExplanation {
+    explanationResult.displayOrder.push('payee', 'forceProxyType');
+    return {
+      ...explanationResult,
+      outputs: [
+        {
+          address: json.controller?.toString() || '',
+          amount: json.amount || '',
+        },
+      ],
+      payee: json.payee,
+      forceProxyType: json.forceProxyType,
+    };
+  }
+
+  explainWalletInitializationTransaction(
+    json: TxData,
+    explanationResult: TransactionExplanation,
+  ): TransactionExplanation {
+    explanationResult.displayOrder.push('owner', 'proxyType', 'delay');
+    return {
+      ...explanationResult,
+      owner: json.owner,
+      proxyType: json.proxyType,
+      delay: json.delay,
+    };
+  }
+
+  explainStakingUnlockTransaction(json: TxData, explanationResult: TransactionExplanation): TransactionExplanation {
+    return {
+      ...explanationResult,
+      outputs: [
+        {
+          address: json.sender.toString(),
+          amount: json.amount || '',
+        },
+      ],
+    };
+  }
+
+  /** @inheritdoc */
+  explainTransaction(): TransactionExplanation {
+    const result = this.toJson();
+    const displayOrder = ['outputAmount', 'changeAmount', 'outputs', 'changeOutputs', 'fee', 'type'];
+    const outputs: TransactionRecipient[] = [];
+    const explanationResult: TransactionExplanation = {
+      // txhash used to identify the transactions
+      id: construct.txHash(this.toBroadcastFormat()),
+      displayOrder,
+      outputAmount: result.amount?.toString() || '0',
+      changeAmount: '0',
+      changeOutputs: [],
+      outputs,
+      fee: {
+        fee: result.tip?.toString() || '',
+        type: 'tip',
+      },
+      type: this.type,
+    };
+    switch (this.type) {
```

Comment:
is addressInitialization a TransactionType for DOT? If so, why is it not included here?
If a new proxy account is set onto a primary account for example, is this an addressInitialization type?

Reply:
WalletIntialization is a type. Used for addProxy transactions to setup proxy accounts.